tools/filesystem/archivefilesystem.go

```diff
@@ -60,12 +60,20 @@ func (s *ArchiveFileSystem) WriteFile(data []byte, fileName string) error {
 }
 
 func (s *ArchiveFileSystem) GenerateTarFileSystemAccess(filePath string) (cpi.BlobAccess, error) {
-	fileInfo, err := s.OsFileSystem.Stat(filePath)
+	tarData, err := GenerateTarData(s.OsFileSystem, filePath)
+	if err != nil {
+		return nil, err
+	}
+	return blobaccess.ForData(tarMediaType, tarData), nil
+}
+
+func GenerateTarData(filesystem vfs.FileSystem, filePath string) ([]byte, error) {
```

Comment:
This can be a private function and tested by its caller func `GenerateTarFileSystemAccess()`.
Could also be a private method, as you like.

Reply:
I extracted this as a separate function on purpose - I don't know how to test it "through" GenerateTarFileSystemAccess, because it wraps the result (a byte slice) in the OCM abstration: `cpi.BlobAccess`.
And to verify if Tar is correctly constructed, I just need a `[]byte`, so even If I learned how to get this data back from `cpi.BlobAccess` I still prefer this explicit and simple solution - my goal here is to verify the Tar byte stream, not to play with OCM stuff.
You are right that it can be a private function so I will make it one, and then test it having the test in the same package as the function (which is totally fine in Go :) )
But I'll have to suppress the linter.

Reply:
Then just inject the dependency you need in this package and test it separately 👍 
Since it's a function you need, you use a func with signature `func(fs vfs.FileSystem, path string) ([]byte, error)` as a required dependency.

Reply:
OK, I refactored it in a bit different way.
I noticed that `GenerateTarFileSystemAccess` method is sort-of alien in the `tools/filesystem/archivefilesystem.go` file.
For example, it introduces dependency to OCM, which doesn't match the apparent purpose of this filesystems.
I removed this dependency and change the function to return just the `[]byte` slice, and renamed it to just: `ArchiveFile`.
Now it's an exported function on the type.
In addition I simplified the `ArchiveFileSystem` interface (it doesn't need anything Tar-related) and I converted few struct fields from exported to unexported - these were exported without a purpose IMO.